resources/tests/test_resources_queries.py

```diff
@@ -562,3 +571,277 @@ def test_get_harbor_available_active_berths(api_client, pier, available):
+            harbors {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = api_client.execute(query)
+    assert executed["data"] == {"harbors": {"count": count, "totalCount": count}}
+
+
+def test_get_harbor_count_filtered(api_client):
+    filtered = random.randint(1, 10)
+    unfiltered = random.randint(1, 10)
+    for _i in range(filtered):
+        harbor = HarborFactory()
+        PierFactory(harbor=harbor, electricity=True)
+    for _i in range(unfiltered):
+        harbor = HarborFactory()
+        PierFactory(harbor=harbor, electricity=False)
+
+    query = """
+        {
+            harbors(piers_Electricity: %s) {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = api_client.execute(query % "true")
+    assert executed["data"] == {
+        "harbors": {"count": filtered, "totalCount": filtered + unfiltered}
+    }
+
+    executed = api_client.execute(query % "false")
+    assert executed["data"] == {
+        "harbors": {"count": unfiltered, "totalCount": filtered + unfiltered}
+    }
+
+
+def test_get_pier_count(api_client):
+    count = random.randint(1, 10)
+    for _i in range(count):
+        PierFactory()
+
+    query = """
+        {
+            piers {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = api_client.execute(query)
+    assert executed["data"] == {"piers": {"count": count, "totalCount": count}}
+
+
+def test_get_pier_count_filtered(api_client):
+    filtered = random.randint(1, 10)
+    unfiltered = random.randint(1, 10)
+    for _i in range(filtered):
+        PierFactory(electricity=True)
+    for _i in range(unfiltered):
+        PierFactory(electricity=False)
+
+    query = """
+        {
+            piers(electricity: %s) {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = api_client.execute(query % "true")
+    assert executed["data"] == {
+        "piers": {"count": filtered, "totalCount": filtered + unfiltered}
+    }
+
+    executed = api_client.execute(query % "false")
+    assert executed["data"] == {
+        "piers": {"count": unfiltered, "totalCount": filtered + unfiltered}
+    }
+
+
+def test_get_berth_count(api_client):
+    count = random.randint(1, 10)
+    for _i in range(count):
+        BerthFactory()
+
+    query = """
+        {
+            berths {
+                count
+                totalCount
+            }
+        }
+    """
+
+    executed = api_client.execute(query)
+    assert executed["data"] == {"berths": {"count": count, "totalCount": count}}
+
+
+def test_get_berth_count_filtered(api_client):
+    filtered = random.randint(1, 10)
+    unfiltered = random.randint(1, 10)
+    filtered_bt = BerthTypeFactory(
+        width=round(random.uniform(0.1, 4.99), 2),
+        length=round(random.uniform(0.1, 4.99), 2),
+        depth=round(random.uniform(0.1, 4.99), 2),
```

Comment:
you could have skipped `length` and `depth`, if you are only filtering by `width` 🙂  would make the test shorter

Reply:
Terrrrrific catch. 👍